src/integer_mod_q/mat_polynomial_ring_zq/get.rs

```diff
@@ -158,6 +158,54 @@ impl GetEntry<PolynomialRingZq> for MatPolynomialRingZq {
 }
 
 impl MatPolynomialRingZq {
+    /// Returns a deep copy of the submatrix defined by the given parameters.
+    /// All entries starting from `(row1, col1)` to `(row2, col2)`(inclusively) are collected in
+    /// a new matrix.
+    /// Note that `row1 >= row2` and `col1 >= col2` must hold. Otherwise the function will panic.
+    ///
+    /// Parameters:
+    /// `row1`: The starting row of the submatrix
+    /// `row2`: The ending row of the submatrix
+    /// `col1`: The starting column of the submatrix
+    /// `col2`: The ending column of the submatrix
+    ///
+    /// Returns the submatrix from `(row1, col1)` to `(row2, col2)`(inclusively).
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer::MatPolyOverZ;
+    /// use qfall_math::integer_mod_q::{MatPolynomialRingZq, ModulusPolynomialRingZq};
+    /// use std::str::FromStr;
+    ///
+    /// let modulus = ModulusPolynomialRingZq::from_str("4  1 0 0 1 mod 17").unwrap();();
+    /// let mat = MatPolyOverZ::identity(3,3);
+    /// let poly_ring_mat = MatPolynomialRingZq::from((&mat, &modulus));
+    ///
+    /// let sub_mat = poly_ring_mat.get_submatrix(0, 2, 1, 1).unwrap();
+    ///
+    /// let e2 = MatPolyOverZ::from_str("[[0],[1  1],[0]]").unwrap();
+    /// let e2 = MatPolynomialRingZq::from((&e2, &modulus));
+    /// assert_eq!(e2, sub_mat)
+    /// ```
+    ///
+    /// # Errors and Failures
+    /// - Returns a [`MathError`] of type [`MathError::OutOfBounds`]
+    /// if any provided row or column is greater than the matrix or negative.
+    ///
+    /// # Panics ...
+    /// - if `col1 > col2` or `row1 > row2`.
+    pub fn get_submatrix(
+        &self,
+        row1: impl TryInto<i64> + Display,
+        row2: impl TryInto<i64> + Display,
+        col1: impl TryInto<i64> + Display,
+        col2: impl TryInto<i64> + Display,
+    ) -> Result<Self, MathError> {
+        Ok(MatPolynomialRingZq {
+            matrix: self.matrix.get_submatrix(row1, row2, col1, col2)?,
+            modulus: self.modulus.clone(),
```

Comment:
Why not self.get_mod()? I think both do the same.

Reply:
done